crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -876,6 +878,11 @@ where
         _ = farmer_cache_worker_fut.fuse() => {
             info!("Farmer cache worker exited.")
         },
+
+        // Prometheus worker future, disabled if there is no prometheus worker
+        Some(_) = prometheus_worker.fuse() => {
```

Comment:
Do we really want to stop the farmer node if the prometheus worker stopped. Shouldn't we just log this and continue instead ?

Prometheus is not an essential task for farmer.

maybe @nazar-pc has a different opinion 🤔 

Also, why cannot we use `spawn_essential` instead? Tokio should exit if the essential task fails anyway

Reply:
> Also, why cannot we use `spawn_essential` instead? Tokio should exit if the essential task fails anyway

Because this is not Substrate :upside_down_face:

Reply:
lol. My bad!